### ISSUE TITLE
platform_lcsense_segmentLCD/src/app.c wrong comments & LCD configuration issue

### ISSUE THREAD

**Issue by silabs-PhilippL** [SiliconLabs/platform_applications#17]
**Project Name**
platform_lcsense_segmentLCD
File: /src/app.c

**Issue Description**
**A)** Within initVDAC() there are a few comment lines that do not reflect the actual configuration.
**B)** The demo enables all 20 LCD segments even though the LCD on the board only supports 10 LCD segments.

**Issues:**

**A.1) Line 177**
Wrong:
`initVdac.reference = vdacRef2V5; // AVDD reference`

This should read:
`initVdac.reference = vdacRef2V5; // internal 2.5 V reference`

**A.2) Line 188 & 189**
According to the configured voltage reference (No. A.1, see above) either the comments are not correct in line 188 or the the configured VDAC voltage in line 189 is not correct.

Wrong:
```
// Set channel 0 output, 2000 / 4095 * 3.3 ~= 1.6V
VDAC_Channel0OutputSet(VDAC0, 3000);
```
This should read:
```
// Set channel 0 output, 3000 / 4096 * 2.5V ~=1.83V
VDAC_Channel0OutputSet(VDAC0, 3000);
```

**B)**
This is probably due to missing flexibility of the SegmentLCD_Init() function (Line 381) which does not allow to mask the non-required LCD segments.
The demo simply enables all LCD segments even though the LCD on the kit only has 10 Segments. This leads to the behavior that GPIOs will be configured and allocated to the LCD even though these GPIOs are connected to other peripherals. This may cause peripheral conflicts and possibly issues on the board.
The LCD that are mistakenly enabled are conflicting with the following board peripherals:
- LED0
- Button1
- PTI
- DBG_SWCLK
- Si7021 (I2C)

Just as an idea how the issue could be solved:
Extend SegmentLCD_Init() (located in lcd driver 'sl_segmentlcd.c' of Gecko SDK) with a parameter that allows to declare the required LCD segments:
E.g.:
`void SegmentLCD_Init(bool useBoost, uint32_t lcdSegmentEnMask)`

Within SegmenLCD_Init() change lines 551-558 to:
```
  for (uint8_t i = 0; i < LCD_COM_LINES_MAX; i++) {
    LCD_ComEnable(i, true);
  }
  for (uint8_t i = 0; i < LCD_SEGMENT_LINES_MAX; i++) {
      if(lcdSegmentEnMask & (1 << i)) {
        LCD_SegmentEnable(i, true);
    }
  }
```
Then, within platform_lcsense_segmentLCD/src/app.c, define a LCD mask:
```
#define LCD_SEGMENT_MASK (uint32_t)((1 << 0)|(1 << 1)|(1 << 4)|(1 << 5)| \
                            (1 << 6)|(1 << 7)|(1 << 8)|(1 << 10)| \
                            (1 << 18)|(1 << 19))
```
and change line 381 as follows:
`SegmentLCD_Init(true, LCD_SEGMENT_MASK);`

**Comment by silabs-Ruibin** [comment]
Hi Philipp, Thank you for reporting this issue! This issue has now been addressed and will be fixed in the next update.

**Comment by silabs-MitchC** [comment]
This issue has been fixed with commit aa6d79d0c0a5fb9b054d7277912af7d863c975bb.  Closing issue.
